
patch landed shortly after 0.13. can be removed once 0.14 is released

From 02aa8c5ae84c779e1eacca9c1610b05bca9d5b8c Mon Sep 17 00:00:00 2001
From: Simon Schubert <2@0x2c.org>
Date: Wed, 5 Feb 2020 12:10:15 +0100
Subject: [PATCH] use strlcpy instead of strncpy

diff -wpruN '--exclude=*.orig' a~/net.c a/net.c
--- a~/net.c	1970-01-01 00:00:00
+++ a/net.c	1970-01-01 00:00:00
@@ -101,7 +101,7 @@ send_remote_command(int fd, const char*
 			s = SSL_get_error(config.ssl, s);
 			if (s != SSL_ERROR_WANT_READ &&
 			    s != SSL_ERROR_WANT_WRITE) {
-				strncpy(neterr, ssl_errstr(), sizeof(neterr));
+				strlcpy(neterr, ssl_errstr(), sizeof(neterr));
 				return (-1);
 			}
 		}
@@ -151,12 +151,12 @@ read_remote(int fd, int extbufsize, char
 			if (((config.features & SECURETRANS) != 0) &&
 			    (config.features & NOSSL) == 0) {
 				if ((rlen = SSL_read(config.ssl, buff + len, sizeof(buff) - len)) == -1) {
-					strncpy(neterr, ssl_errstr(), sizeof(neterr));
+					strlcpy(neterr, ssl_errstr(), sizeof(neterr));
 					goto error;
 				}
 			} else {
 				if ((rlen = read(fd, buff + len, sizeof(buff) - len)) == -1) {
-					strncpy(neterr, strerror(errno), sizeof(neterr));
+					strlcpy(neterr, strerror(errno), sizeof(neterr));
 					goto error;
 				}
 			}
